### PR TITLE
File preview of comment attachments

### PR DESCRIPTION
**Pull request by kmshin1397**
- Supports PDF, JPG, PNG, GIF

preview

- Adjusted comment attachment test accordingly, but couldn't get it to be robust enough to remove xfail on Travis. Test passes locally for me.

### CODE REVIEW

**Review by acrellin — APPROVED** [review]
Looks good, thanks @kmshin1397  -- just a few questions below

File: skyportal/app_server.py [hunk 1]
```diff
@@ -96,6 +96,9 @@ def make_app(cfg, baselayer_handlers, baselayer_settings):
         (r'/api/classification(/[0-9]+)?', ClassificationHandler),
         (r'/api/comment(/[0-9]+)?', CommentHandler),
         (r'/api/comment(/[0-9]+)/attachment', CommentAttachmentHandler),
+        # Allow the '.pdf' suffix for the attachment route, as the react-file-previewer
+        # package expects URLs ending with '.pdf' to load PDF files.
```
[the remark this answers is not in the document]
Comment (kmshin1397) [reply skyportal/app_server.py:100]: Yeah, all the other supported file types are images and just passed along to an  tag I think and can follow the same logic. The ".pdf" url ending is how the package determines whether to render a <canvas /> for a PDF or just assume it's an image file.

File: static/js/components/CommentAttachmentPreview.jsx [hunk 2]
```diff
@@ -0,0 +1,221 @@
...
+  const handleClose = () => {
+    setOpen(false);
+  };
+
+  const fileType = filename.includes(".") ? filename.split(".", 2)[1] : "";
+  const knownType = ["png", "jpg", "jpeg", "pdf", "gif"].includes(
+    fileType.toLowerCase()
+  );
+
+  // The FilePreviewer expects a url ending with .pdf for PDF files
```
[the remark this answers is not in the document]
Comment (kmshin1397) [reply static/js/components/CommentAttachmentPreview.jsx:127]: The other alternative I found was [this](https://www.npmjs.com/package/react-file-viewer), which is definitely more fully-featured, but also much much larger (181 KB vs 11.8 MB). I think it's better to hack this one a bit then to import another 11MB for functionality we'll never need.

File: static/js/components/CommentAttachmentPreview.jsx [hunk 3]
```diff
@@ -0,0 +1,221 @@
...
+            <button
+              type="button"
+              className={classes.linkButton}
+              onClick={handleClickOpen}
+              data-testid={`attachmentButton_${filename.split(".")[0]}`}
+            >
+              {shortenFilename(filename)}
+            </button>
+          </div>
+        </Tooltip>
```
[the remark this answers is not in the document]
Comment (kmshin1397) [reply static/js/components/CommentAttachmentPreview.jsx]: Done

File: static/js/components/CommentAttachmentPreview.jsx [hunk 4]
```diff
@@ -0,0 +1,221 @@
...
+  const handleClickOpen = () => {
+    setOpen(true);
+  };
+
+  const handleClose = () => {
+    setOpen(false);
+  };
+
+  const fileType = filename.includes(".") ? filename.split(".", 2)[1] : "";
+  const knownType = ["png", "jpg", "jpeg", "pdf", "gif"].includes(
```
[the remark this answers is not in the document]
Comment (kmshin1397) [reply static/js/components/CommentAttachmentPreview.jsx]: Done

**Comment by stefanv** [comment]
Nice one, @kmshin1397

**Pull request merged**